eblocker-icapserver/src/main/java/org/eblocker/server/common/blacklist/CacheIndex.java

```diff
@@ -16,26 +16,94 @@
  */
 package org.eblocker.server.common.blacklist;
 
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
 class CacheIndex {
-    private int format;
-    private Map<Integer, List<CachedFileFilter>> filters;
+    private final int format;
+    private final Map<Integer, List<CachedFileFilter>> filters;
 
-    public int getFormat() {
-        return format;
+    CacheIndex(int format) {
+        this.format = format;
+        filters = new HashMap<>();
     }
 
-    public void setFormat(int format) {
+    CacheIndex(@JsonProperty("format") int format, @Nonnull @JsonProperty("filters") Map<Integer, List<CachedFileFilter>> filters) {
         this.format = format;
+        this.filters = filters;
+    }
+
+    int getFormat() {
+        return format;
     }
 
-    public Map<Integer, List<CachedFileFilter>> getFilters() {
```

Comment:
Here I would like to protect the hashmap from unwanted changes. As I have seen, the organisation of the hashmap is of key importance.